server/services/organisationService.ts

```diff
@@ -2,33 +2,66 @@ import createError from 'http-errors'
 import type { ResponseError } from 'superagent'
 
 import type { PrisonClient, RestClientBuilder } from '../data'
-import { OrganisationUtils } from '../utils'
+import { OrganisationUtils, TypeUtils } from '../utils'
 import type { Organisation } from '@accredited-programmes/models'
+import type { Prison } from '@prison-register-api'
 
 export default class OrganisationService {
   constructor(private readonly prisonClientBuilder: RestClientBuilder<PrisonClient>) {}
 
-  async getOrganisation(token: Express.User['token'], organisationId: string): Promise<Organisation | null> {
+  async getOrganisation(token: Express.User['token'], organisationId: Organisation['id']): Promise<Organisation> {
     const prisonClient = this.prisonClientBuilder(token)
 
     try {
       const prison = await prisonClient.find(organisationId)
 
       if (!prison.active) {
-        return null
+        throw createError(500, `Organisation with ID ${organisationId} not active.`)
```

Comment:
I'm struggling to figure out what we should return here. What made you go for a 500? I think it's probably the most appropriate, I was tempted by a 422, or even a 404 but it's a bit of a hard one to pin down.

Reply:
Shall we go with 404 as a first step? I think you're right in saying the thing we're trying to find is active organisations (although that's not all that explicit from the URL) but I think, as you say, the message makes it clear.

Reply:
I wonder if the message should be `Active organisation with ID ... not found`?

Reply:
Yeah that sounds good to me